packages/full-stack-tests/src/components/unified-selection/PropertyGrid.test.tsx

```diff
@@ -0,0 +1,130 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+
+import { expect } from "chai";
+import React from "react";
+import { UiComponents, VirtualizedPropertyGridWithDataProvider } from "@itwin/components-react";
+import { IModelApp, IModelConnection } from "@itwin/core-frontend";
+import { useDisposable } from "@itwin/core-react";
+import { InstanceKey, KeySet } from "@itwin/presentation-common";
+import { PresentationPropertyDataProvider, usePropertyDataProviderWithUnifiedSelection } from "@itwin/presentation-components";
+import { Presentation } from "@itwin/presentation-frontend";
+import { buildTestIModel } from "@itwin/presentation-testing";
+import { act, fireEvent, render, waitFor } from "@testing-library/react";
+import { insertPhysicalElement, insertPhysicalModel, insertSpatialCategory } from "../../IModelUtils";
+import { initialize, terminate } from "../../IntegrationTests";
+
+/* eslint-disable @typescript-eslint/naming-convention */
+
+describe("Learning snippets", async () => {
+
+  describe("Unified selection", () => {
+
+    describe("Property grid", () => {
+
+      before(async () => {
+        await initialize();
+        await UiComponents.initialize(IModelApp.localization);
+      });
+
+      after(async () => {
+        await terminate();
+      });
+
+      it("renders unified selection property grid", async function () {
+        // __PUBLISH_EXTRACT_START__ Presentation.Components.UnifiedSelection.PropertyGrid
+        function MyPropertyGrid(props: { imodel: IModelConnection }) {
+          // create a presentation rules driven data provider; the provider implements `IDisposable`, so we
+          // create it through `useDisposable` hook to make sure it's properly cleaned up
+          const dataProvider = useDisposable(React.useCallback(() => {
```

Comment:
There is no requirement to `import React from "react"` for JSX to work so I would prefer importing only react types and function that are used. ex: `import { useCallback } from "react"`

Reply:
I thinks there is no need to show that it's a `React` function as our components package is made specifically for `React` so I hoper package consumer knows about hooks used in this example. 

I want to have consistent imports inside whole repository, but if you think it gives some value to the example we can leave it.

Reply:
> I want to have consistent imports inside whole repository

makes sense, updated